control-plane/cni/config/config.go

```diff
@@ -0,0 +1,18 @@
+package config
+
+type CNIConfig struct {
+	// Name of the plugin
+	Name string `json:"name" mapstructure:"name"`
+	// Type of plugin (consul-cni)
+	Type string `json:"type" mapstructure:"type"`
+	// CNIBinDir is the location of the cni config files on the node. Can bet as a cli flag.
+	CNIBinDir string `json:"cni_bin_dir" mapstructure:"cni_bin_dir"`
+	// CNINetDir is the locaion of the cni plugin on the node. Can be set as a cli flag.
+	CNINetDir string `json:"cni_net_dir" mapstructure:"cni_net_dir"`
+	// Multus is if the plugin is a multus plugin. Can be set as a cli flag.
+	Multus bool `json:"multus" mapstructure:"multus"`
+	// Kubeconfig file name. Can be set as a cli flag.
+	Kubeconfig string `json:"kubeconfig" mapstructure:"kubeconfig"`
+	// LogLevl is the logging level. Can be set as a cli flag.
+	LogLevel string `json:"log_level" mapstructure:"log_level"`
+}
```

Comment:
This struct is used by both the installer and plugin. It is a representation of the config file that is written to the host.